spec/serialization_spec.rb

```diff
@@ -43,6 +45,505 @@ def self.name
+
+        it 'coerces integers' do
+          PORO::Employee.create(age: 40)
+          render
+          expect(attributes['age']).to eq('0.4E2')
+        end
+
+        it 'coerces strings' do
+          PORO::Employee.create(age: '40.01')
+          render
+          expect(attributes['age']).to eq('0.4001E2')
+        end
+
+        it 'allows nils' do
+          PORO::Employee.create(age: nil)
+          render
+          expect(attributes['age']).to eq(nil)
+        end
+
+        # Unfortunate we don't get a better error class
+        context 'when cannot coerce' do
+          before do
+            PORO::Employee.create(age: {})
+          end
+
+          it 'raises error' do
+            expect {
+              render
+            }.to raise_error(NoMethodError)
+          end
+        end
+      end
+
+      context 'when float' do
+        before do
+          resource.attribute :age, :float
+        end
+
+        it 'coerces strings' do
+          PORO::Employee.create(age: '40.01')
+          render
+          expect(attributes['age']).to eq(40.01)
+        end
+
+        it 'coerces integers' do
+          PORO::Employee.create(age: 40)
+          render
+          expect(attributes['age']).to eq(40.0)
+        end
+
+        it 'allows nils' do
+          PORO::Employee.create(age: nil)
+          render
+          expect(attributes['age']).to eq(nil)
+        end
+
+        context 'when cannot coerce' do
+          before do
+            PORO::Employee.create(age: {})
+          end
+
+          it 'raises error' do
+            expect {
+              render
+            }.to raise_error(TypeError)
+          end
+        end
+      end
+
+      context 'when boolean' do
+        before do
+          resource.attribute :age, :boolean
+        end
+
+        it 'coerces strings' do
+          PORO::Employee.create(age: 'true')
+          render
+          expect(attributes['age']).to eq(true)
+        end
+
+        it 'coerces integers' do
+          PORO::Employee.create(age: 1)
+          render
+          expect(attributes['age']).to eq(true)
+        end
+
+        it 'allows nils' do
+          PORO::Employee.create(age: nil)
+          render
+          expect(attributes['age']).to eq(nil)
+        end
+
+        context 'when cannot coerce' do
+          before do
+            PORO::Employee.create(age: 1.1)
+          end
+
+          it 'raises error' do
+            expect {
+              render
+            }.to raise_error(Dry::Types::ConstraintError)
+          end
+        end
+      end
+
+      context 'when date' do
+        before do
+          resource.attribute :age, :date
+        end
+
+        it 'coerces Date to correct string format' do
+          PORO::Employee.create(age: Date.parse('2018/01/06'))
+          render
+          expect(attributes['age']).to eq('2018-01-06')
+        end
+
+        it 'coerces Time to correct date string format' do
+          time = Time.parse('2018/01/06 4:13pm')
```

Comment:
Assuming it uses local server timezone to determine the correct date, but may want to add specs to better define the behavior.

Reply:
There are specs for this.